semanticdb/semanticdb/src/main/scala/scala/meta/internal/semanticdb/Scala.scala

```diff
@@ -282,9 +282,15 @@ object Scala {
   }
 
   private[meta] object DescriptorParser {
-    def apply(symbol: String): (Descriptor, String) = {
+    private final val _cache = new java.util.HashMap[String, (Descriptor, String)]()
```

Comment:
I'm on the fence about introducing global caches in library code like here, we have had a lot of problems with a similar cache that is used by the tokenizer https://github.com/scalameta/scalameta/issues/1068. I don't believe this cache will have the same correctness issues as the tokenizer cache but there is still a risk of growing memory usage.

I believe it's safer to introduce caches in applications like rsc/scalafmt/scalafix/metals where there are clear phases for cleaning up acquired resources.

Is this descriptor parser used by rsc?

Reply:
That's fair. This is used by rsc. I would then consider a way to easily override the included descriptor parser if the user wanted to have a cache, how does that sound?